### PR TITLE
sign up feature added

### PR DESCRIPTION
**Pull request by flutterindiadev**

### CODE REVIEW

**Comment by flutterindiadev** [comment]
Adding new pull request

File: core/domain/lib/src/usecase/base/base_usecase.dart [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import 'package:dartz/dartz.dart';
-import 'package:shared/shared.dart';
 import 'params.dart';
 
-abstract class BaseUseCase<E extends BaseError, P extends Params, T> {
+abstract class BaseUseCase<E, P extends Params, T> {
```
Comment (naikdp7) [inline core/domain/lib/src/usecase/base/base_usecase.dart]: it requires E extends BaseError

File: core/domain/lib/src/usecase/user/signup_usercase.dart [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  final UserRepository _userRepository;
+
+  SignUpUsecase(this._userRepository);
+
+  @override
+  Future<Either<BaseError, User>> execute(
+      {required SignupParams params}) async {
+    return Future.value(
+      (await _userRepository.signUpWithEmail(params.signUpRequest))
+      //     .fold((l) => Left(l), (result) async {
```
Comment (naikdp7) [inline core/domain/lib/src/usecase/user/signup_usercase.dart]: what is the reason for comment?
it should implement saveUser as well.

File: core/shared/lib/src/model/request/sign_up_request.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+part 'sign_up_request.g.dart';
+
+@JsonSerializable(explicitToJson: true)
+class SignUpRequest {
+  String email;
+  String password;
+  factory SignUpRequest.fromJson(Map<String, dynamic> json) =>
+      _$SignUpRequestFromJson(json);
+
+  Map<String, dynamic> toJson() => _$SignUpRequestToJson(this);
```
Comment (naikdp7) [inline core/shared/lib/src/model/request/sign_up_request.dart]: this should not have the toJson method as well as fromJson

Comment (naikdp7) [reply core/shared/lib/src/model/request/sign_up_request.dart]: you need to create a signUpApiRequest class and that class will have this from and to json methods

File: core/shared/lib/src/model/signup_request.dart [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// import 'signup_provider.dart';
+
+// class SignUpRequest {
```
Comment (naikdp7) [inline core/shared/lib/src/model/signup_request.dart]: why we need this class and why is it commented?

File: infrastructure/network-retrofit/lib/src/request/sign_up_request.dart [hunk 5]
```diff
@@ -0,0 +1,15 @@
+// import 'package:json_annotation/json_annotation.dart';
+
+// part 'sign_up_request.g.dart';
+
+// @JsonSerializable(explicitToJson: true)
```
Comment (naikdp7) [inline infrastructure/network-retrofit/lib/src/request/sign_up_request.dart]: remove this if its unused.

File: infrastructure/network-retrofit/lib/src/services/retrofit_service.dart [hunk 6]
```diff
@@ -8,4 +10,8 @@ abstract class RetrofitService {
   factory RetrofitService(Dio dio, {String? baseUrl}) {
     return _RetrofitService(dio, baseUrl: dio.options.baseUrl);
   }
+  @POST("signup")
+  Future<HttpResponse<UserResponseEntity>> signUp(
+    @Body() SignUpRequest signUpRequest,
```
Comment (naikdp7) [inline infrastructure/network-retrofit/lib/src/services/retrofit_service.dart:15]: this should be SignUpApiRequest

**Review by naikdp7 — COMMENTED** [review]
Please remove .idea folder

**Review by naikdp7 — APPROVED** [review]

**Pull request merged**